maskhit/train.py

```diff
@@ -123,27 +136,66 @@ def get_resume_checkpoint(checkpoints_name, epoch_to_resume):
     return checkpoint_to_resume
 
 
-def prepare_data(meta_split, meta_file, round_month=False, vars_to_include=[]):
+def prepare_data(meta_split, meta_file, vars_to_include=[]):
+    """
+    Merge and preprocess meta_split and meta_file dataframes for use by the model.
+
+    Parameters
+    ----------
+    meta_split : pandas.DataFrame
+        A pandas dataframe containing the split information for each patient. The dataframe must contain information about patient ids
+
+    meta_file : pandas.DataFrame
+        A pandas dataframe containing the patient metadata. The dataframe must contain a column named 'id_patient'
+
+    vars_to_include : list, optional
+        A list of additional variables to include in the merged dataframe. Default is an empty list.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A merged and preprocessed pandas dataframe containing the split information and patient metadata. The returned dataframe may contain the following columns:
+        - id_patient: unique patient ID
+        - case_number: patient case number (may need to be corrected - showing as 'SP' for IBD dataset)
+        - id_patient_num: encoded patient ID
+        - id_svs_num: encoded id_svs
+        - outcome: patient outcome variable, encoded for classification models e.g. 0, 1, 2 for three classes
+
+    """
+
+    if 'id_patient' not in meta_split.columns:
+        patient_ids = []
+        # iterating over the meta_split dataframe
+        for index, row in meta_split.iterrows():
+            # obtaining the paths of the files to the related slide
+            file_names = ast.literal_eval(row['Path'])
+            patient_id = file_names[0].split('/')[5].split(' ')[0]
+            patient_ids.append(patient_id) # adding patient id to the list
+        meta_split['id_patient'] = patient_ids # adding column to the meta_split dataframe
+        # formatting rows in meta_file of the id patients so they match that of meta_split df
+        meta_file['id_patient'] = meta_file['id_patient'].apply(lambda x: pd.Series(x.split(' ')[0]))
+
+    print(meta_split) # contains 'id_patient'

```

Comment:
When you print something, please make sure adding some explanation. 
1. Tagging: [INFO], [DEBUG], [ERROR] etc
2. One line to explain what's printed.